rust/android/Dockerfile

```diff
@@ -25,6 +22,11 @@ RUN apt-get update \
  && unzip /tmp/android-ndk.zip -d /usr/local/
 
 
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs -o ./rustup.sh
```

Comment:
Nice! Glad to see this `curl` on its own line, before running it.

I know this is the preferred method, and this doesn't have to be blocking, but I'm going to pick on `curl`. I considered...
* downgrade attacks : documentation seems to state this should be covered by the `--proto '=https'` and `--tlsv1.2`, but I didn't test it myself.
* certificate pinning : this is largely deprecated in practice, and I'm not sure if this would work at all. But if we use `curl`, a syntax like `--pinnedpubkey "sha256//73d34t…"` might work, if the site's TLS supports it. If we stick with `curl` I think we should try it.
* certificate status: consider `--cert-status`. Apparently `curl` has no check for certificate revocation or expiration by default. We're using `ca-certificates` just a few lines above, could we leverage that from the OS? However, "--cert-status only checks for the status_request TLS extension"  so that might not be a viable technique either.
* signature validation : I don't think there is anything in place currently to either "manually" inspect the `.sh` file or cryptographically validate it.
* nameserver manipulation" consider use of `--doh-url <URL>`, with which we could specify a nameserver and use encryption for the name resolution.

I feel like with `curl` there's a lot of fiddly things with ad-hoc or unclear fixes... Possible other techniques, besides `curl`, found on [a page linked from main language installation page,](https://forge.rust-lang.org/infra/other-installation-methods.html)  :
* installing via OS package. We're using Debian so this should be in place, and come with at least package integrity checks. I don't know how fast these are to update though, I know Debian can be kind of slow at times.
* install by way of "manually" handling `rustup-init`. Links to various architectures are on the same page.
* standalone installers. These (tarballs) look pretty good to me as they can be verified against Rust GPG keys. 
* build from source. Seems cumbersome, if we could work with some of the above points in another way.

Again, this might be something to ignore for the simple "happy path" alreay established, or for futrue refinements.